### PR TITLE
initial version of rest-contracts-lambda

### PR DESCRIPTION
**Pull request by UppaJung**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The REST-Contracts set of packages were designed for web services that run TypeS
    - with types for each function's parameters so that TypeScript can enforce the contract
    - with each function returning a Promise for the result type specified in the contract
  * use TypeScript to simplify implementing the API service while reducing errors
-   - parameters arrive to your code with the types specified in the contract
+   - parameters arrive to your code with the typeps specified in the contract
```
Comment (pawsong) [inline README.md]: This might be a typo

File: packages/rest-contracts-lambda/package.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        "branches": 80,
+        "functions": 80,
+        "lines": 80,
+        "statements": 85
+      }
+    }
+  },
+  "dependencies": {
+    "@types/aws-lambda": "8.10.3",
+    "rest-contracts": "0.9.12"
```
Comment (pawsong) [inline packages/rest-contracts-lambda/package.json]: How about make this a peer-dependency?

Similar case: https://github.com/apollographql/react-apollo/blob/master/package.json#L90

File: packages/rest-contracts-lambda/src/rest-contracts-lambda.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  try {
+    const result = await handler( event, context );
+    callback(undefined, {
+      // tslint:disable:no-magic-numbers
+      statusCode:
+        (api.method === RestContracts.Method.get && typeof(result) === "undefined") ? 404 :
+        (api.method === RestContracts.Method.put) ? 201 :
+        200,
+      // tslint:enable:no-magic-numbers
+      body: JSON.stringify(result)
```
Comment (pawsong) [inline packages/rest-contracts-lambda/src/rest-contracts-lambda.ts]: Could we ensure that result is JSON string?

**Review by pawsong — COMMENTED** [review]
My 2 cents :)

File: README.md (hunk 1 above)
Comment (UppaJung) [reply README.md]: Fixed

File: packages/rest-contracts-lambda/package.json (hunk 2 above)
Comment (UppaJung) [reply packages/rest-contracts-lambda/package.json]: will do.

File: packages/rest-contracts-lambda/src/rest-contracts-lambda.ts (hunk 3 above)
Comment (UppaJung) [reply packages/rest-contracts-lambda/src/rest-contracts-lambda.ts]: I think you are asking to allow multiple types of result types: not just JSON, but raw strings or possibly even binary that would be base64 encoded?  I just checked in a change that will do just that.

While I was add it, I added CORS support.

**Pull request merged**